homeassistant/components/cover/device_action.py

```diff
@@ -0,0 +1,199 @@
+"""Provides device automations for Cover."""
+from typing import List, Optional
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_SUPPORTED_FEATURES,
+    CONF_DEVICE_ID,
+    CONF_DOMAIN,
+    CONF_ENTITY_ID,
+    CONF_TYPE,
+    SERVICE_CLOSE_COVER,
+    SERVICE_CLOSE_COVER_TILT,
+    SERVICE_OPEN_COVER,
+    SERVICE_OPEN_COVER_TILT,
+    SERVICE_SET_COVER_POSITION,
+    SERVICE_SET_COVER_TILT_POSITION,
+    SERVICE_STOP_COVER,
+    SERVICE_STOP_COVER_TILT,
+)
+from homeassistant.core import Context, HomeAssistant
+from homeassistant.helpers import entity_registry
+import homeassistant.helpers.config_validation as cv
+
+from . import (
+    ATTR_POSITION,
+    ATTR_TILT_POSITION,
+    DOMAIN,
+    SUPPORT_CLOSE,
+    SUPPORT_CLOSE_TILT,
+    SUPPORT_OPEN,
+    SUPPORT_OPEN_TILT,
+    SUPPORT_SET_POSITION,
+    SUPPORT_SET_TILT_POSITION,
+    SUPPORT_STOP,
+    SUPPORT_STOP_TILT,
+)
+
+CMD_ACTION_TYPES = {"open", "close", "stop", "open_tilt", "close_tilt", "stop_tilt"}
+POSITION_ACTION_TYPES = {"set_position", "set_tilt_position"}
```

Comment:
I am wondering if instead of open/close/set_position, we just add a single device action: change position. Same with tilt.

Reply:
Simplified now; if set_position is supported only expose a set_position action. Likewise, if set_tilt_position is supported only expose a set_tilt_position action.